streaming/base/partition/orig.py

```diff
@@ -81,7 +81,8 @@ def get_partitions_orig(num_samples: int,
             padding = node_ratio - overflow
     padded_samples_per_canonical_node = samples_per_canonical_node + padding
 
-    if num_samples > num_canonical_nodes:
+    if num_samples > num_canonical_nodes or (num_samples == num_canonical_nodes and
+                                             num_canonical_nodes > num_physical_nodes):
```

Comment:
I believe the condition here can actually be slightly expanded to `num_canonical nodes >= num_physical nodes`. So the full conditional would be:
```
if num_samples > num_canonical_nodes or (num_samples == num_canonical_nodes and num_canonical_nodes >= num_physical_nodes):
```
Mind making this change and adding an explanatory comment right before the if statement?

Reply:
Thank you for the input; I've made that change now.